lib/browser/api/web-contents.ts

```diff
@@ -579,6 +570,30 @@ WebContents.prototype._init = function () {
     enumerable: true
   });
 
+  /**
+   * Cached IPC emitters sorted by dispatch priority.
+   * Caching is used to avoid frequent array allocations.
+   *
+   * 0: WebFrameMain ipc
+   * 1: WebContents ipc
+   * 2: ipcMain
+   */
+  const cachedIpcEmitters: (ElectronInternal.IpcMainInternal | undefined)[] = new Array(3);
```

Comment:
nit: we can initialize the array with the contents to avoid the extra step
```suggestion
  const cachedIpcEmitters: (ElectronInternal.IpcMainInternal | undefined)[] = [undefined, ipc, ipcMain];
```

Reply:
Preallocating to a fixed size was a preoptimization. I looked into it and your suggestion is better to avoid creating "holey" arrays. 🙇  https://v8.dev/blog/elements-kinds#avoid-creating-holes